lib/tlscontext.c

```diff
@@ -335,6 +338,52 @@ _set_sni_in_client_mode(TLSSession *self)
   return FALSE;
 }
 
+void
+_write_line_to_keylog_file(const char *file_path, const char *line, FILE *keylog_file, GMutex *mutex)
+{
+  gint ret_val;
+  g_mutex_lock(mutex);
+  if(keylog_file)
+    {
+      ret_val = fprintf(keylog_file, "%s\n", line);
+      if (ret_val != strlen(line))
```

Comment:
The `printf` above writes the `line` plus a newline to the file, as such the `ret_val` should be `strlen(line)+1`.